### ISSUE TITLE
Confusion between builtin `undefined` and enum variant

### ISSUE THREAD

**Issue by CoffeeImpliesCode** [DonIsaac/zlint#242]
**Version**
main @ https://github.com/DonIsaac/zlint/commit/83d016d6d12b4563fc58a40060c1623facb791ac  (after 0.6.3)

**Describe the bug**
The linter gets confused between the builtin `undefined` keyword and an enum variant named `undefined` (which is perfectly legal), wanting me to add a safety comment.

**To Reproduce**
* create `enum` with variant `.undefined`
* use `zlint`
* receive the following warning:
```bash
  ⚠ unsafe-undefined: `undefined` is missing a safety comment
   ╭─[src/main.zig:4:18]
 4 │ const A = enum { undefined, hello };
   ·                  ─────────
   ╰────
  help: Add a `SAFETY: <reason>` before this line explaining why this code is safe.
```

**Expected behavior**
there should be no safety lint triggered on the enum A. 
 
**Screenshots**
see `To Reproduce`

**Additional Context**
This is unhelpful, as the API spec I'm working with defines enums with the value `undefined`, so changing it is not really a preferred option. I currently have ~30 warnings of this sort.
